app/main.py

```diff
@@ -1,13 +1,19 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return ({"line" if key == "line_number"
+            else "column" if key == "column_number"
+            else "message" if key == "text"
+            else "name" if key == "code" else key: value
+            for key, value in error.items()
+            if key not in ("filename", "physical_line")}
+            | {"source": "flake8"})
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return ({"errors": [format_linter_error(error) for error in errors]}
+            | {"path": file_path}
+            | {"status": "passed" if len(errors) == 0 else "failed"})
```

Comment:
Do not join several dicts, create one dict with the required keys at once. Besides that, do not place the dict inside of the `()`, and do not forget that an empty list is faulty by default, so there is no need to check its length.